==165659== Memcheck, a memory error detector
==165659== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==165659== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==165659== Command: target/x86_64-unknown-linux-gnu/release/njvm tests/data/v8/trivial.bin
==165659== Parent PID: 165657
==165659== 
==165659== Syscall param statx(file_name) points to unaddressable byte(s)
==165659==    at 0x49B888E: statx (statx.c:29)
==165659==    by 0x131EE1: statx (weak.rs:178)
==165659==    by 0x131EE1: std::sys::unix::fs::try_statx (fs.rs:150)
==165659==    by 0x128A50: file_attr (fs.rs:851)
==165659==    by 0x128A50: metadata (fs.rs:494)
==165659==    by 0x128A50: std::fs::buffer_capacity_required (fs.rs:610)
==165659==    by 0x128973: read_to_end (fs.rs:638)
==165659==    by 0x128973: std::fs::read::inner (fs.rs:232)
==165659==    by 0x115A00: njvm::utils::read_file::read_file (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1129FB: njvm::NinjaVM<R,W>::load_binary (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x11392F: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==165659== 
==165659== Syscall param statx(buf) points to unaddressable byte(s)
==165659==    at 0x49B888E: statx (statx.c:29)
==165659==    by 0x131EE1: statx (weak.rs:178)
==165659==    by 0x131EE1: std::sys::unix::fs::try_statx (fs.rs:150)
==165659==    by 0x128A50: file_attr (fs.rs:851)
==165659==    by 0x128A50: metadata (fs.rs:494)
==165659==    by 0x128A50: std::fs::buffer_capacity_required (fs.rs:610)
==165659==    by 0x128973: read_to_end (fs.rs:638)
==165659==    by 0x128973: std::fs::read::inner (fs.rs:232)
==165659==    by 0x115A00: njvm::utils::read_file::read_file (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1129FB: njvm::NinjaVM<R,W>::load_binary (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x11392F: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==165659== 
==165659== 
==165659== HEAP SUMMARY:
==165659==     in use at exit: 9,478 bytes in 10 blocks
==165659==   total heap usage: 22 allocs, 12 frees, 12,593 bytes allocated
==165659== 
==165659== 8 bytes in 1 blocks are definitely lost in loss record 1 of 9
==165659==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==165659==    by 0x112E89: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659== 
==165659== 8 bytes in 1 blocks are definitely lost in loss record 2 of 9
==165659==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==165659==    by 0x112EB1: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659== 
==165659== 25 bytes in 1 blocks are still reachable in loss record 3 of 9
==165659==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==165659==    by 0x1285C7: alloc (alloc.rs:89)
==165659==    by 0x1285C7: alloc_impl (alloc.rs:171)
==165659==    by 0x1285C7: allocate (alloc.rs:231)
==165659==    by 0x1285C7: allocate_in<u8, alloc::alloc::Global> (raw_vec.rs:185)
==165659==    by 0x1285C7: with_capacity_in<u8, alloc::alloc::Global> (raw_vec.rs:132)
==165659==    by 0x1285C7: with_capacity_in<u8, alloc::alloc::Global> (mod.rs:613)
==165659==    by 0x1285C7: to_vec<u8, alloc::alloc::Global> (slice.rs:227)
==165659==    by 0x1285C7: to_vec<u8, alloc::alloc::Global> (slice.rs:176)
==165659==    by 0x1285C7: to_vec_in<u8, alloc::alloc::Global> (slice.rs:507)
==165659==    by 0x1285C7: to_vec<u8> (slice.rs:482)
==165659==    by 0x1285C7: {closure#0} (args.rs:146)
==165659==    by 0x1285C7: {closure#0}<isize, std::ffi::os_str::OsString, (), std::sys::unix::args::imp::clone::{closure_env#0}, core::iter::traits::iterator::Iterator::for_each::call::{closure_env#0}<std::ffi::os_str::OsString, alloc::vec::spec_extend::{impl#1}::spec_extend::{closure_env#0}<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::alloc::Global>>> (map.rs:84)
==165659==    by 0x1285C7: fold<core::ops::range::Range<isize>, (), core::iter::adapters::map::map_fold::{closure_env#0}<isize, std::ffi::os_str::OsString, (), std::sys::unix::args::imp::clone::{closure_env#0}, core::iter::traits::iterator::Iterator::for_each::call::{closure_env#0}<std::ffi::os_str::OsString, alloc::vec::spec_extend::{impl#1}::spec_extend::{closure_env#0}<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::alloc::Global>>>> (iterator.rs:2362)
==165659==    by 0x1285C7: fold<std::ffi::os_str::OsString, core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}, (), core::iter::traits::iterator::Iterator::for_each::call::{closure_env#0}<std::ffi::os_str::OsString, alloc::vec::spec_extend::{impl#1}::spec_extend::{closure_env#0}<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::alloc::Global>>> (map.rs:124)
==165659==    by 0x1285C7: for_each<core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::vec::spec_extend::{impl#1}::spec_extend::{closure_env#0}<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::alloc::Global>> (iterator.rs:779)
==165659==    by 0x1285C7: spec_extend<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::alloc::Global> (spec_extend.rs:40)
==165659==    by 0x1285C7: from_iter<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>> (spec_from_iter_nested.rs:62)
==165659==    by 0x1285C7: from_iter<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>> (spec_from_iter.rs:33)
==165659==    by 0x1285C7: from_iter<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>> (mod.rs:2554)
==165659==    by 0x1285C7: collect<core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::vec::Vec<std::ffi::os_str::OsString, alloc::alloc::Global>> (iterator.rs:1784)
==165659==    by 0x1285C7: clone (args.rs:143)
==165659==    by 0x1285C7: args (args.rs:128)
==165659==    by 0x1285C7: args (args.rs:19)
==165659==    by 0x1285C7: std::env::args_os (env.rs:802)
==165659==    by 0x1284B0: std::env::args (env.rs:767)
==165659==    by 0x11307F: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659== 
==165659== 40 bytes in 1 blocks are still reachable in loss record 4 of 9
==165659==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==165659==    by 0x12DEF6: alloc (alloc.rs:89)
==165659==    by 0x12DEF6: alloc_impl (alloc.rs:171)
==165659==    by 0x12DEF6: allocate (alloc.rs:231)
==165659==    by 0x12DEF6: exchange_malloc (alloc.rs:320)
==165659==    by 0x12DEF6: new<std::sys::unix::locks::pthread_mutex::Mutex> (boxed.rs:195)
==165659==    by 0x12DEF6: from<std::sys::unix::locks::pthread_mutex::Mutex> (boxed.rs:1386)
==165659==    by 0x12DEF6: std::sys_common::mutex::MovableMutex::new (mutex.rs:64)
==165659==    by 0x12D587: new<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>> (mutex.rs:218)
==165659==    by 0x12D587: {closure#0} (stdio.rs:324)
==165659==    by 0x12D587: {closure#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::io::stdio::stdin::{closure_env#0}> (lazy.rs:281)
==165659==    by 0x12D587: {closure#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::lazy::{impl#10}::get_or_init::{closure_env#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::io::stdio::stdin::{closure_env#0}>, !> (lazy.rs:462)
==165659==    by 0x12D587: std::sync::once::Once::call_once_force::{{closure}} (once.rs:334)
==165659==    by 0x10FFC9: std::sync::once::Once::call_inner (once.rs:434)
==165659==    by 0x11014A: call_once_force<std::lazy::{impl#10}::initialize::{closure_env#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::lazy::{impl#10}::get_or_init::{closure_env#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::io::stdio::stdin::{closure_env#0}>, !>> (once.rs:334)
==165659==    by 0x11014A: std::lazy::SyncOnceCell<T>::initialize (lazy.rs:461)
==165659==    by 0x129BDB: get_or_try_init<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::lazy::{impl#10}::get_or_init::{closure_env#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::io::stdio::stdin::{closure_env#0}>, !> (lazy.rs:328)
==165659==    by 0x129BDB: get_or_init<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::io::stdio::stdin::{closure_env#0}> (lazy.rs:281)
==165659==    by 0x129BDB: std::io::stdio::stdin (stdio.rs:323)
==165659==    by 0x112E76: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659== 
==165659== 48 bytes in 1 blocks are still reachable in loss record 5 of 9
==165659==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==165659==    by 0x128540: alloc (alloc.rs:89)
==165659==    by 0x128540: alloc_impl (alloc.rs:171)
==165659==    by 0x128540: allocate (alloc.rs:231)
==165659==    by 0x128540: allocate_in<std::ffi::os_str::OsString, alloc::alloc::Global> (raw_vec.rs:185)
==165659==    by 0x128540: with_capacity_in<std::ffi::os_str::OsString, alloc::alloc::Global> (raw_vec.rs:132)
==165659==    by 0x128540: with_capacity_in<std::ffi::os_str::OsString, alloc::alloc::Global> (mod.rs:613)
==165659==    by 0x128540: with_capacity<std::ffi::os_str::OsString> (mod.rs:470)
==165659==    by 0x128540: from_iter<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>> (spec_from_iter_nested.rs:54)
==165659==    by 0x128540: from_iter<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>> (spec_from_iter.rs:33)
==165659==    by 0x128540: from_iter<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>> (mod.rs:2554)
==165659==    by 0x128540: collect<core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::vec::Vec<std::ffi::os_str::OsString, alloc::alloc::Global>> (iterator.rs:1784)
==165659==    by 0x128540: clone (args.rs:143)
==165659==    by 0x128540: args (args.rs:128)
==165659==    by 0x128540: args (args.rs:19)
==165659==    by 0x128540: std::env::args_os (env.rs:802)
==165659==    by 0x1284B0: std::env::args (env.rs:767)
==165659==    by 0x112F73: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659== 
==165659== 69 bytes in 2 blocks are still reachable in loss record 6 of 9
==165659==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==165659==    by 0x1285C7: alloc (alloc.rs:89)
==165659==    by 0x1285C7: alloc_impl (alloc.rs:171)
==165659==    by 0x1285C7: allocate (alloc.rs:231)
==165659==    by 0x1285C7: allocate_in<u8, alloc::alloc::Global> (raw_vec.rs:185)
==165659==    by 0x1285C7: with_capacity_in<u8, alloc::alloc::Global> (raw_vec.rs:132)
==165659==    by 0x1285C7: with_capacity_in<u8, alloc::alloc::Global> (mod.rs:613)
==165659==    by 0x1285C7: to_vec<u8, alloc::alloc::Global> (slice.rs:227)
==165659==    by 0x1285C7: to_vec<u8, alloc::alloc::Global> (slice.rs:176)
==165659==    by 0x1285C7: to_vec_in<u8, alloc::alloc::Global> (slice.rs:507)
==165659==    by 0x1285C7: to_vec<u8> (slice.rs:482)
==165659==    by 0x1285C7: {closure#0} (args.rs:146)
==165659==    by 0x1285C7: {closure#0}<isize, std::ffi::os_str::OsString, (), std::sys::unix::args::imp::clone::{closure_env#0}, core::iter::traits::iterator::Iterator::for_each::call::{closure_env#0}<std::ffi::os_str::OsString, alloc::vec::spec_extend::{impl#1}::spec_extend::{closure_env#0}<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::alloc::Global>>> (map.rs:84)
==165659==    by 0x1285C7: fold<core::ops::range::Range<isize>, (), core::iter::adapters::map::map_fold::{closure_env#0}<isize, std::ffi::os_str::OsString, (), std::sys::unix::args::imp::clone::{closure_env#0}, core::iter::traits::iterator::Iterator::for_each::call::{closure_env#0}<std::ffi::os_str::OsString, alloc::vec::spec_extend::{impl#1}::spec_extend::{closure_env#0}<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::alloc::Global>>>> (iterator.rs:2362)
==165659==    by 0x1285C7: fold<std::ffi::os_str::OsString, core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}, (), core::iter::traits::iterator::Iterator::for_each::call::{closure_env#0}<std::ffi::os_str::OsString, alloc::vec::spec_extend::{impl#1}::spec_extend::{closure_env#0}<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::alloc::Global>>> (map.rs:124)
==165659==    by 0x1285C7: for_each<core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::vec::spec_extend::{impl#1}::spec_extend::{closure_env#0}<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::alloc::Global>> (iterator.rs:779)
==165659==    by 0x1285C7: spec_extend<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::alloc::Global> (spec_extend.rs:40)
==165659==    by 0x1285C7: from_iter<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>> (spec_from_iter_nested.rs:62)
==165659==    by 0x1285C7: from_iter<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>> (spec_from_iter.rs:33)
==165659==    by 0x1285C7: from_iter<std::ffi::os_str::OsString, core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>> (mod.rs:2554)
==165659==    by 0x1285C7: collect<core::iter::adapters::map::Map<core::ops::range::Range<isize>, std::sys::unix::args::imp::clone::{closure_env#0}>, alloc::vec::Vec<std::ffi::os_str::OsString, alloc::alloc::Global>> (iterator.rs:1784)
==165659==    by 0x1285C7: clone (args.rs:143)
==165659==    by 0x1285C7: args (args.rs:128)
==165659==    by 0x1285C7: args (args.rs:19)
==165659==    by 0x1285C7: std::env::args_os (env.rs:802)
==165659==    by 0x1284B0: std::env::args (env.rs:767)
==165659==    by 0x112F73: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659== 
==165659== 536 bytes in 1 blocks are still reachable in loss record 7 of 9
==165659==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==165659==    by 0x112A1E: njvm::NinjaVM<R,W>::load_binary (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x11392F: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659== 
==165659== 552 bytes in 1 blocks are still reachable in loss record 8 of 9
==165659==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==165659==    by 0x119BF3: alloc::raw_vec::finish_grow (raw_vec.rs:0)
==165659==    by 0x10FA8B: grow_amortized<u8, alloc::alloc::Global> (raw_vec.rs:400)
==165659==    by 0x10FA8B: alloc::raw_vec::RawVec<T,A>::reserve::do_reserve_and_handle (raw_vec.rs:285)
==165659==    by 0x128987: reserve<u8, alloc::alloc::Global> (raw_vec.rs:289)
==165659==    by 0x128987: reserve<u8, alloc::alloc::Global> (mod.rs:813)
==165659==    by 0x128987: read_to_end (fs.rs:638)
==165659==    by 0x128987: std::fs::read::inner (fs.rs:232)
==165659==    by 0x115A00: njvm::utils::read_file::read_file (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1129FB: njvm::NinjaVM<R,W>::load_binary (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x11392F: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659== 
==165659== 8,192 bytes in 1 blocks are still reachable in loss record 9 of 9
==165659==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==165659==    by 0x12D55A: alloc (alloc.rs:89)
==165659==    by 0x12D55A: alloc_impl (alloc.rs:171)
==165659==    by 0x12D55A: allocate (alloc.rs:231)
==165659==    by 0x12D55A: allocate_in<u8, alloc::alloc::Global> (raw_vec.rs:185)
==165659==    by 0x12D55A: with_capacity_in<u8, alloc::alloc::Global> (raw_vec.rs:132)
==165659==    by 0x12D55A: with_capacity<u8> (raw_vec.rs:93)
==165659==    by 0x12D55A: new_uninit_slice<u8> (boxed.rs:616)
==165659==    by 0x12D55A: with_capacity<std::io::stdio::StdinRaw> (bufreader.rs:96)
==165659==    by 0x12D55A: {closure#0} (stdio.rs:324)
==165659==    by 0x12D55A: {closure#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::io::stdio::stdin::{closure_env#0}> (lazy.rs:281)
==165659==    by 0x12D55A: {closure#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::lazy::{impl#10}::get_or_init::{closure_env#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::io::stdio::stdin::{closure_env#0}>, !> (lazy.rs:462)
==165659==    by 0x12D55A: std::sync::once::Once::call_once_force::{{closure}} (once.rs:334)
==165659==    by 0x10FFC9: std::sync::once::Once::call_inner (once.rs:434)
==165659==    by 0x11014A: call_once_force<std::lazy::{impl#10}::initialize::{closure_env#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::lazy::{impl#10}::get_or_init::{closure_env#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::io::stdio::stdin::{closure_env#0}>, !>> (once.rs:334)
==165659==    by 0x11014A: std::lazy::SyncOnceCell<T>::initialize (lazy.rs:461)
==165659==    by 0x129BDB: get_or_try_init<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::lazy::{impl#10}::get_or_init::{closure_env#0}<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::io::stdio::stdin::{closure_env#0}>, !> (lazy.rs:328)
==165659==    by 0x129BDB: get_or_init<std::sync::mutex::Mutex<std::io::buffered::bufreader::BufReader<std::io::stdio::StdinRaw>>, std::io::stdio::stdin::{closure_env#0}> (lazy.rs:281)
==165659==    by 0x129BDB: std::io::stdio::stdin (stdio.rs:323)
==165659==    by 0x112E76: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659== 
==165659== LEAK SUMMARY:
==165659==    definitely lost: 16 bytes in 2 blocks
==165659==    indirectly lost: 0 bytes in 0 blocks
==165659==      possibly lost: 0 bytes in 0 blocks
==165659==    still reachable: 9,462 bytes in 8 blocks
==165659==         suppressed: 0 bytes in 0 blocks
==165659== 
==165659== ERROR SUMMARY: 4 errors from 4 contexts (suppressed: 0 from 0)
==165659== 
==165659== 1 errors in context 1 of 4:
==165659== Syscall param statx(buf) points to unaddressable byte(s)
==165659==    at 0x49B888E: statx (statx.c:29)
==165659==    by 0x131EE1: statx (weak.rs:178)
==165659==    by 0x131EE1: std::sys::unix::fs::try_statx (fs.rs:150)
==165659==    by 0x128A50: file_attr (fs.rs:851)
==165659==    by 0x128A50: metadata (fs.rs:494)
==165659==    by 0x128A50: std::fs::buffer_capacity_required (fs.rs:610)
==165659==    by 0x128973: read_to_end (fs.rs:638)
==165659==    by 0x128973: std::fs::read::inner (fs.rs:232)
==165659==    by 0x115A00: njvm::utils::read_file::read_file (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1129FB: njvm::NinjaVM<R,W>::load_binary (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x11392F: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==165659== 
==165659== 
==165659== 1 errors in context 2 of 4:
==165659== Syscall param statx(file_name) points to unaddressable byte(s)
==165659==    at 0x49B888E: statx (statx.c:29)
==165659==    by 0x131EE1: statx (weak.rs:178)
==165659==    by 0x131EE1: std::sys::unix::fs::try_statx (fs.rs:150)
==165659==    by 0x128A50: file_attr (fs.rs:851)
==165659==    by 0x128A50: metadata (fs.rs:494)
==165659==    by 0x128A50: std::fs::buffer_capacity_required (fs.rs:610)
==165659==    by 0x128973: read_to_end (fs.rs:638)
==165659==    by 0x128973: std::fs::read::inner (fs.rs:232)
==165659==    by 0x115A00: njvm::utils::read_file::read_file (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1129FB: njvm::NinjaVM<R,W>::load_binary (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x11392F: njvm::main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114A52: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x114DE8: _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h96decfe9dd09f250E.llvm.10521280659363928151 (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==    by 0x1274DD: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (function.rs:259)
==165659==    by 0x1274DD: do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panicking.rs:492)
==165659==    by 0x1274DD: try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32> (panic.rs:137)
==165659==    by 0x1274DD: {closure#2} (rt.rs:128)
==165659==    by 0x1274DD: do_call<std::rt::lang_start_internal::{closure_env#2}, isize> (panicking.rs:492)
==165659==    by 0x1274DD: try<isize, std::rt::lang_start_internal::{closure_env#2}> (panicking.rs:456)
==165659==    by 0x1274DD: catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize> (panic.rs:137)
==165659==    by 0x1274DD: std::rt::lang_start_internal (rt.rs:128)
==165659==    by 0x114001: main (in /home/clemens/Projects/LowLevelProjects/RustProjects/njvm/target/x86_64-unknown-linux-gnu/release/njvm)
==165659==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==165659== 
==165659== ERROR SUMMARY: 4 errors from 4 contexts (suppressed: 0 from 0)
